src/views/Rules/ListRules/index.jsx

```diff
@@ -557,17 +714,18 @@ function ListRules(props) {
         open={dialogState.open}
         title={dialogState.title}
         destructive={dialogState.destructive}
-        body={dialogBody}
+        body={dialogMode === 'delete' ? deleteDialogBody : signoffDialogBody}
         confirmText={dialogState.confirmText}
-        onSubmit={handleDialogSubmit}
+        onSubmit={() => dialogState.handleSubmit(dialogState)}
         onError={handleDialogError}
         error={dialogState.error}
-        onComplete={handleDialogComplete}
+        onComplete={dialogState.handleComplete}
         onClose={handleDialogClose}
+        onExited={handleDialogExited}
```

Comment:
Setting `open: false` sounds like the right thing to do. I wonder if we actually need the `onExited` function. The UI behavior seems to be unchanged whether `handleDialogExited` is called or not. There's also no problem leaving it here. Your call.

Reply:
Just for cleanliness, I think it's probably best to clean up the state, so I'll keep it in.